src/main/java/org/broadinstitute/hellbender/tools/exome/ExomeReadCounts.java

```diff
@@ -0,0 +1,881 @@
+     */
+    public static final String ANONYMOUS_SAMPLE_ID = "<anon-sample>";
+
+    /**
+     * Default read-group id used when the actual read-group id could not be resolved.
+     */
+    public static final String ANONYMOUS_READ_GROUP_ID = "<anon-readgroup>";
+
+    /**
+     * Full name for the {@link #groupBy} argument.
+     */
+    protected static final String GROUP_BY_FULL_NAME = "groupBy";
+
+    /**
+     * Short name for the {@link #groupBy} argument.
+     */
+    protected static final String GROUP_BY_SHORT_NAME = GROUP_BY_FULL_NAME;
+
+    /**
+     * Short name for the {@link #columnSummaryOutput} argument.
+     */
+    protected static final String COLUMN_SUMMARY_OUTPUT_SHORT_NAME = "CSO";
+
+    /**
+     * Full name for the {@link #columnSummaryOutput} argument.
+     */
+    protected static final String COLUMN_SUMMARY_OUTPUT_FULL_NAME = "columnSummaryOutput";
+
+    /**
+     * Short name for the {@link #rowSummaryOutput} argument.
+     */
+    protected static final String ROW_SUMMARY_OUTPUT_SHORT_NAME = "RSO";
+
+    /**
+     * Full name for the {@link #rowSummaryOutput} argument.
+     */
+    protected static final String ROW_SUMMARY_OUTPUT_FULL_NAME = "rowSummaryOutput";
+
+    /**
+     * Full name for the {@link #cohortName} argument.
+     */
+    protected static final String COHORT_FULL_NAME = "cohortName";
+
+    /**
+     * Short name for the {@link #cohortName} argument.
+     */
+    protected static final String COHORT_SHORT_NAME = "shortName";
+
+    /**
+     * Full name for the {@link #countColumnName} argument.
+     */
+    protected static final String COUNT_COLUMN_FULL_NAME = "countColumnName";
+
+    /**
+     * Short name for the {@link #countColumnName} argument.
+     */
+    protected static final String COUNT_COLUMN_SHORT_NAME = COUNT_COLUMN_FULL_NAME;
+
+    /**
+     * Format string for the column total average per bp.
+     */
+    protected static final String OUTPUT_DOUBLE_FORMAT = "%.4f";
+
+    /**
+     * Possible read count grouping.
+     */
+    protected enum GroupBy {
+
+        /**
+         * Single read count for all input.
+         */
+        COHORT(CohortCountColumns::new),
+
+        /**
+         * Count per sample.
+         */
+        SAMPLE(SampleCountColumns::new),
+
+        /**
+         * Count per read-group.
+         */
+        READ_GROUP(ReadGroupCountColumns::new);
+
+        /**
+         * Corresponding count-columns manager factory.
+         */
+        private final Function<ExomeReadCounts,? extends CountColumns> countColumnsFactory;
+
+        /**
+         * Creates a new group-by option.
+         * @param countColumnsFactory the count-columns manager factory to be used with
+         *                            this group-by option. Assumed not to be {@code null}.
+         */
+        GroupBy(final Function<ExomeReadCounts,? extends CountColumns> countColumnsFactory) {
```

Comment:
The enum is always static and thus I cannot make reference to the ExomeReadCounts.this